plugin/src/main/java/com/denizenscript/denizen/objects/WorldTag.java

```diff
@@ -1006,6 +1008,92 @@ else if (time >= 12500) {
+        // Returns whether the ender dragon has been killed in this world before.
+        // Only works in end worlds.
+        // -->
+        registerTag(ElementTag.class, "first_dragon_killed", (attribute, object) -> {
+            DragonBattle battle = object.getWorld().getEnderDragonBattle();
+            if (battle == null) {
+                attribute.echoError("Provided world is not an end world!");
+                return null;
+            }
+            return new ElementTag(battle.hasBeenPreviouslyKilled());
+        });
+
+        // <--[mechanism]
+        // @object WorldTag
+        // @name respawn_dragon
+        // @description
+        // Initiates the respawn sequence of the ender dragon as if a player placed 4 end crystals on the portal.
+        // Only works in end worlds.
+        // -->
+        tagProcessor.registerMechanism("respawn_dragon", false, (object, mechanism) -> {
+            DragonBattle battle = object.getWorld().getEnderDragonBattle();
+            if (battle == null) {
+                mechanism.echoError("Provided world is not an end world!");
+                return;
+            }
+            battle.initiateRespawn();
+        });
+
+        // <--[mechanism]
+        // @object WorldTag
+        // @name reset_crystals
+        // @description
+        // Resets the end crystals located on the obsidian pillars in this world.
+        // Only works in end worlds.
+        // -->
+        tagProcessor.registerMechanism("reset_crystals", false, (object, mechanism) -> {
+            DragonBattle battle = object.getWorld().getEnderDragonBattle();
+            if (battle == null) {
+                mechanism.echoError("Provided world is not an end world!");
+                return;
+            }
+            battle.resetCrystals();
+        });
+
+        // <--[mechanism]
+        // @object WorldTag
+        // @name respawn_phase
+        // @input ElementTag
+        // @description
+        // Set the current respawn phase of the ender dragon. Valid phases can be found at <@link url https://jd.papermc.io/paper/1.20/org/bukkit/boss/DragonBattle.RespawnPhase.html>
+        // Only works in end worlds.
+        // -->
+        tagProcessor.registerMechanism("respawn_phase", false, ElementTag.class, (object, mechanism, input) -> {
+            DragonBattle battle = object.getWorld().getEnderDragonBattle();
+            if (battle == null) {
+                mechanism.echoError("Provided world is not an end world!");
+                return;
+            }
+            battle.setRespawnPhase(input.asEnum(DragonBattle.RespawnPhase.class));
+        });
+
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
+
+            // <--[mechanism]
+            // @object WorldTag
+            // @name respawn_phase
+            // @input ElementTag(Boolean)
+            // @description
+            // Set whether the first ender dragon was killed already.
+            // Only works in end worlds.
```

Comment:
Might be nice to document exactly what this does - does it reset the end and respawn the dragon? is it just a value that doesn't really affect anything?